test/jdk/sun/security/tools/keytool/i18n.java

```diff
@@ -23,17 +23,410 @@
+    private Thread currentThread = null;
+
+    public static class DialogBuilder {
+        private JDialog dialog;
+        private JTextArea instructionsText;
+        private JTextArea messageText;
+        private JButton pass;
+        private JButton fail;
+
+        public DialogBuilder() {
+            dialog = new JDialog(new JFrame());
+            dialog.setDefaultCloseOperation(JDialog.DO_NOTHING_ON_CLOSE);
+            instructionsText = new JTextArea("", 5, 100);
+
+            dialog.add("North", new JScrollPane(instructionsText,
+                    JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+                    JScrollPane.HORIZONTAL_SCROLLBAR_ALWAYS));
+
+            messageText = new JTextArea("", 20, 100);
+            dialog.add("Center", new JScrollPane(messageText,
+                    JScrollPane.VERTICAL_SCROLLBAR_ALWAYS,
+                    JScrollPane.HORIZONTAL_SCROLLBAR_ALWAYS));
+
+            JPanel buttons = new JPanel();
+            pass = new JButton("pass");
+            pass.setActionCommand("pass");
+            buttons.add("East", pass);
+
+            fail = new JButton("fail");
+            fail.setActionCommand("fail");
+            buttons.add("West", fail);
+
+            dialog.add("South", buttons);
+        }
+
+        public DialogBuilder setTitle(String title) {
+            dialog.setTitle(title);
+            return this;
+        }
+
+        public DialogBuilder setInstruction(String instruction) {
+            instructionsText.setText("Test instructions:\n" + instruction);
+            return this;
+        }
+
+        public DialogBuilder setMessage(String message) {
+            messageText.setText(message);
+            return this;
+        }
+
+        public DialogBuilder setPassAction(ActionListener action) {
+            pass.addActionListener(action);
+            return this;
+        }
+
+        public DialogBuilder setFailAction(ActionListener action) {
+            fail.addActionListener(action);
+            return this;
+        }
+
+        public DialogBuilder setAbortAction(Runnable action) {
+            dialog.addWindowListener(new WindowAdapter() {
+                @Override
+                public void windowClosing(WindowEvent e) {
+                    super.windowClosing(e);
+                    action.run();
+                }
+            });
+            return this;
+        }
+
+        public JDialog build() {
+            dialog.pack();
+            return dialog;
+        }
+    }
+
+    public static void executeKeytool(String command) throws Exception {
+        sun.security.tools.keytool.Main.main(command.split("\\s+"));
+    }
+
+    public static void main(String[] args) {
+        if (args.length == 1) {
+            Locale.setDefault(Locale.of(args[0]));
+        } else if (args.length == 2) {
+            Locale.setDefault(Locale.of(args[0], args[1]));
+        }
+        final String LANG = Locale.getDefault().getDisplayLanguage();
```

Comment:
Instead of `getDisplayLanguage()`, it should issue `getDisplayName()`, as for `zh-CN` case, it simply displays `Chinese` in the current impl. It's ambiguous whether it is simplified or traditional.
Also, `LANG` should be lowercase, as it is not a constant.